src/main/java/de/tum/in/www1/exerciseapp/service/AuthorizationCheckService.java

```diff
@@ -0,0 +1,51 @@
+package de.tum.in.www1.exerciseapp.service;
+
+import de.tum.in.www1.exerciseapp.domain.Authority;
+import de.tum.in.www1.exerciseapp.domain.Course;
+import de.tum.in.www1.exerciseapp.domain.Exercise;
+import de.tum.in.www1.exerciseapp.domain.User;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service used to check whether user is authorized to perform actions on the entity.
+ */
+@Service
+public class AuthorizationCheckService {
+
+    private final UserService userService;
+    private Authority adminAuthority;
+
+    public AuthorizationCheckService(UserService userService) {
+        this.userService = userService;
+        adminAuthority = new Authority();
+        adminAuthority.setName("ROLE_ADMIN");
+    }
+
+    public boolean isAuthorizedForCourse(Course course) {
+        if(course == null) {
+            return true;
+        }
+        User user = userService.getUserWithGroupsAndAuthorities();
+        if(user.getGroups().contains(course.getStudentGroupName())
+            || user.getGroups().contains(course.getTeachingAssistantGroupName())
+            || user.getGroups().contains(course.getInstructorGroupName())
+            || user.getAuthorities().contains(adminAuthority)
+            || course.getTitle().equals("Archive")) {
```

Comment:
I am not sure where you got this code from. This seems to be old code that is not used any more... please have a look into the master branch

Reply:
at least the "Archive" option should not be used any more

Reply:
The "Archive" option is used at least in the findAll() method in the ExerciseService, that's why I believed it was relevant. The rest are from the findAll() method of CourseService, which you showed me yourself.